File: leetcode/java/RemoveDuplicatesfromSortedArray.java
class Solution {
    public int removeDuplicates(int[] nums) {
        if(nums == null) throw new IllegalArgumentException("x");
        int i = 0;
        int j = i+1;
        while(j < nums.length) {
            if(nums[i] == nums[j]) j++;
            else nums[++i] = nums[j++];
        }
        return i+1;
    }
}

File: leetcode/c++/Remove_Duplicates_from_Sorted_Array.cpp
class Solution {
public:
    int removeDuplicates(int A[], int n) {
        if(n <= 1) return n;
        
        int i = 0;
        int j = 1;
        
        while(j < n){
            if(A[i] == A[j]) j++;
            else A[++i] = A[j++];
        }

        return i+1;
    }
};
